tests_e2e/tests/lib/exception.py

```diff
@@ -0,0 +1,28 @@
+from typing import Any
```

Comment:
We shouldn't have an exception module with all exceptions. Exceptions should be defined in the module that raises them, does CommandError need to be here?

If you need to refactor out the other 2 execeptions to a separate module, I'd suggest naming it "TestResult" or something similar.

Reply:
if it is raised by multiple modules, it can be refactored to a separate file.

why does CommandError need to be moved?

Reply:
I thought of circular-import issue at some point and moved here; I'll check and keep it in original place if its not craeaing issue now